### ISSUE TITLE
Some problems about checkerboard artifacts

### ISSUE THREAD

**Issue by huangzehao** [alexjc/neural-enhance#26]
Hi, alexjc! Great repo for super-resolution!
I have some problems on how to handle the checkerboard artifacts when using deconvolution-like techniques.

(1) As we know, one way to handle the checkerboard artifactes is handle an resized image by a convolutional neural network, like SRCNN. This is a general approach in super-resolution. But it's slow since the size of input is huge based on the scale.
(2) Another way is using deconvolution-like techniques, like subpixel convolution, or upsampling convolution.
I tested subpixel convolution and nearest-upsampling convolution in torch. It seems that there are not obvious differences in the results. But both of them have strange black and white pixel blocks like following image.
![21077](https://cloud.githubusercontent.com/assets/13661914/19846214/2eadddc8-9f78-11e6-8ae0-6b5bce0588a4.jpg)

I haven't met this situation when I train SR like (1)[[This is my implementation of VDSR]](https://github.com/huangzehao/caffe-vdsr).

My network is the MSE-SRResNet in paper Photo-Realistic Single Image Super-Resolution Using a Generative Adversarial Network. This is my [model config](https://gist.github.com/huangzehao/32474b23e9903a07589c72624dd6a6b1)
I trained on imagenet validset, which contains 50k images, 300 epochs with 0.0001 learning rate, 0.9 beta1. 

Do you have any idea about how to handle this problem?
Have you checked the performance of just MSE based SRResNet? Thank you!
BTW, what's the training data in your training? Maybe I should train a new MSE-SRResNet by them based on your code.

**Comment by alexjc** [comment]
I don't have definite answers for you, just a few observations:
1) High-frequency detail patterns are often desirable, you want the NN to inject these.
2) The GAN does a very good job of reducing these checkerboard artifacts, like TV loss but better.
3) I have bad experiences with nearest-neighbor rescaling, it requires more layers to fix pixelation.
4) I train on 300k images from OpenImages in Flickr, more images help.
5) I don't use MSE at all and haven't compared the results, only conv2_2 perceptual loss.

**Comment by huangzehao** [comment]
Thanks for your experiences! 
I have some questions about your generator network configuration. 
Why you set filter sizes in first and last layer bigger than 3*3? Is this helpful in someway?
What about batch normalization layers? I found that there are not BN layers in the generator. 
Thanks!

**Comment by alexjc** [comment]
@huangzehao The large filter sizes are inspired by Justin's perceptual losses paper (reference [1] in the README). They seem to help, yes, as do additional layers.

I removed BN as it made training less stable, and I believe it's a hack and mistake to use them in generator networks, as you want to maintain absolute values/thresholds. (Unlike discriminators that benefit from BN.) Also, using PRELU improves training so BN is less critical.

**Comment by huangzehao** [comment]
Thanks for you help!

**Comment by AbbasEl-Bourji** [comment]
Hello, Please go to Play Store and update the App, in the new update
version do you can download photos, videos and documents. If you update the
App you DON’T need to rescan the QR code again!  And if you have android
version 5.0 and up then the size of the app is after the update only 9 mb.
Please give us 5 stars and a good review my friend. Have fun with the App !
